tools/tesseratos/src/tesseratos/console/plugin.cpp

```diff
@@ -0,0 +1,290 @@
+#include <imgui.h>
+
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/imgui/plugin.hpp>
+
+#include <tesseratos/console/plugin.hpp>
+#include <tesseratos/toolbox/plugin.hpp>
+
+using cubos::engine::Cubos;
+
+using namespace tesseratos;
+
+namespace
+{
+    struct Mode
+    {
+        std::size_t prevCursor = 0; // Previous cursor position~could find a better way to clear log
+        bool clearEnabled = false;
+        bool collapseEnabled = false;
+        bool traceEnabled = true;
+        bool debugEnabled = true;
+        bool infoEnabled = true;
+        bool warnEnabled = true;
+        bool errorEnabled = true;
+        bool criticalEnabled = true;
+    };
+} // namespace
+
+static std::string levelToString(cubos::core::Logger::Level level)
+{
+    switch (level)
+    {
+    case cubos::core::Logger::Level::Trace:
+        return "Trace";
+    case cubos::core::Logger::Level::Debug:
+        return "Debug";
+    case cubos::core::Logger::Level::Info:
+        return "Info";
+    case cubos::core::Logger::Level::Warn:
+        return "Warn";
+    case cubos::core::Logger::Level::Error:
+        return "Error";
+    case cubos::core::Logger::Level::Critical:
+        return "Critical";
+    default:
+        return "Unknown";
+    }
+}
+
+static ImVec4 levelToColor(cubos::core::Logger::Level level)
+{
+    switch (level)
+    {
+    case cubos::core::Logger::Level::Trace:
+        return ImVec4(1.0F, 1.0F, 1.0F, 1.0F); // White
+    case cubos::core::Logger::Level::Debug:
+        return ImVec4(0.0F, 1.0F, 1.0F, 1.0F); // Cyan
+    case cubos::core::Logger::Level::Info:
+        return ImVec4(0.0F, 1.0F, 0.0F, 1.0F); // Green
+    case cubos::core::Logger::Level::Warn:
+        return ImVec4(1.0F, 1.0F, 0.0F, 1.0F); // Yellow
+    case cubos::core::Logger::Level::Error:
+        return ImVec4(1.0F, 0.0F, 0.0F, 1.0F); // Red
+    case cubos::core::Logger::Level::Critical:
+        return ImVec4(1.0F, 1.0F, 1.0F, 1.0F); // White (or any color you prefer for critical)
+    default:
+        return {1.0F, 1.0F, 1.0F, 1.0F}; // Default to
+    }
+}
+
+void tesseratos::consolePlugin(Cubos& cubos)
+{
+    cubos.depends(cubos::engine::imguiPlugin);
+    cubos.depends(toolboxPlugin);
+    cubos.resource<Mode>();
+
+    cubos.system("show Logger UI").tagged(cubos::engine::imguiTag).call([](Toolbox& toolbox, Mode& mode) {
+        if (!toolbox.isOpen("Console"))
+        {
+            return;
+        }
+
+        static char searchString[256];
+
+        ImGui::SetNextWindowSize(ImVec2(930, 310), ImGuiCond_FirstUseEver);
+        ImGui::SetNextWindowPos(ImVec2(10, 10), ImGuiCond_FirstUseEver);
+        ImGui::Begin("Console");
+
+        ImGui::Columns(2, "ConsoleLayout", true);
+
+        // Log Scrollable Window
+        ImGui::SetColumnWidth(0, ImGui::GetWindowWidth() * 0.8f);
+
+        std::size_t cursor = mode.prevCursor;
+        cubos::core::Logger::Entry entry;
+        std::vector<cubos::core::Logger::Entry> ui_entries;
```

Comment:
This vector should be stored in a resource (Mode?), no need to recreate it every frame.